exporter/debugexporter/exporter.go

```diff
@@ -60,6 +71,22 @@ func (s *debugExporter) pushTraces(_ context.Context, td ptrace.Traces) error {
 		return nil
 	}
 
+	for _, rs := range td.ResourceSpans().All() {
```

Comment:
This changes the Debug exporter to one that [mutates data](https://github.com/open-telemetry/opentelemetry-collector/blob/v0.129.0/exporter/README.md#data-ownership). This comes with a performance penalty - the whole telemetry batch may need to be copied in memory. I'd rather keep the Debug exporter as non-mutating. Would it be possible to perform the filtering when formatting output?